core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -380,6 +380,17 @@ class SparkContext(config: SparkConf) extends Logging {
 
     val resourcesFileOpt = conf.get(DRIVER_RESOURCES_FILE)
     _resources = getOrDiscoverAllResources(_conf, SPARK_DRIVER_PREFIX, resourcesFileOpt)
+    // driver submitted in client mode under Standalone may have conflict resources with
+    // workers on this host. We should sync driver's resources info into SPARK_RESOURCES
+    // to avoid collision.
+    if (deployMode == "client" && (master.startsWith("spark://")
+      || master.startsWith("local-cluster"))) {
+      val requests = parseAllResourceRequests(_conf, SPARK_DRIVER_PREFIX).map {req =>
+        req.id.resourceName -> req.amount
+      }.toMap
+      // TODO(wuyi) log driver's acquired resources separately ?
```

Comment:
Hi, @Ngone51 . Please don't use user-id TODO in the patch. As you know, Apache Spark repository has already a few ancient user-id TODOs like this which is not fixed until now. :)
Since we don't know the future, let's use JIRA-IDed TODO like `TODO(SPARK-XXX)`.

Reply:
@dongjoon-hyun Thank you for reminding that. I'll fix those TODOs in following commits.